lib/stream/leo-stream.js

```diff
@@ -212,15 +212,15 @@ module.exports = function(configure) {
 
 			args.push(ls.through((obj, done) => {
 				var e;
-				if (opts.autoDetectPayload && obj.payload) {
+				if (opts.autoDetectPayload && (obj.payload || obj.correlation_id)) {
```

Comment:
This is so that if you send "checkpoint" events that don't have a payload it doesn't try and wrap it in the payload